spinalcordtoolbox/scripts/sct_qc.py

```diff
@@ -81,16 +81,10 @@ def main(argv: Sequence[str]):
     set_loglevel(verbose=verbose)
 
     from spinalcordtoolbox.reports.qc import generate_qc
-    # Build args list (for display)
-    args_disp = '-i ' + arguments.i
-    if arguments.d:
-        args_disp += ' -d ' + arguments.d
-    if arguments.s:
-        args_disp += ' -s ' + arguments.s
     generate_qc(fname_in1=arguments.i,
                 fname_in2=arguments.d,
                 fname_seg=arguments.s,
-                args=args_disp,
+                args=f'("sct_qc {" ".join(argv)}")',
```

Comment:
I'm probably just being paranoid, but how about using `spinalcordtoolbox.utils.list2cmdline` here? It's what `generate_qc` would end up doing if we passed a list instead of a string, and it would take care of shell quoting any weird arguments. (This would also need an extra import at the top of the file.)
```suggestion
                args=f'(sct_qc {list2cmdline(argv)})',
```

Reply:
In truth I hadn't realized this function existed! 

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/842b0422f1b6f1743be42d366927e1af5c7a9f8f/spinalcordtoolbox/utils/sys.py#L364-L365

Looking at [`shlex.quote`](https://docs.python.org/3/library/shlex.html#shlex.quote), it seems to be designed to add safety to  strings that will be executed as command line arguments. My hunch was that it was originally designed for `run_proc`, given its usage there?

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/842b0422f1b6f1743be42d366927e1af5c7a9f8f/spinalcordtoolbox/utils/sys.py#L385-L388

_(In fact, we should probably be using `shlex.quote` directly on `cmd`, too!)_

But, since we're not executing anything in the context of QC reports, I'm wondering if using this function is overkill?

Reply:
> It's what `generate_qc` would end up doing if we passed a list instead of a string, and it would take care of shell quoting any weird arguments.

Although, as you say, `generate_qc` does use this function too... I wonder why? Does it have a meaningful effect there?

I'm wondering if `list2cmdline` should just be inlined in `run_proc`, and QC should be using `" ".join`?

Reply:
> I'm wondering if `list2cmdline` should removed, with the functionality inlined in `run_proc`, and `generate_qc` should be using `" ".join` instead? Just to convey that `shlex.quote` should only be used in the context of commands to execute.

I don't think that would work. The most common "weird" character handled by `shlex.quote` is a space character. So, calling `shlex.quote` before or after `" ".join` would have a drastically different effect! For example:
```bash
$ args() { printf 'got %s arguments:\n' "$#"; printf '|%s|\n' "$@"; }
$ args 'a' 'b' ' c'
got 3 arguments:
|a|
|b|
| c|
$ args 'a b  c'
got 1 arguments:
|a b  c|
```

Reply:
> But, since we're not executing anything in the context of QC reports (we're just echoing what was passed to the command line), I'm wondering if using this function is overkill? It might even mislead the user as to what was actually passed as arguments?

That's fair. If we don't intend/expect the command to be copy-pasted into a terminal, then maybe we can be careless with the quoting. The only downside is that doing `" ".join` on the arguments loses information in the (unlikely?) case that an argument actually contained a space character.

Reply:
> So, calling `shlex.quote` before or after `" ".join` would have a drastically different effect!

Ah, my apologies. I think I may have miscommunicated? My suggestion was:

- Replace `list2cmdline` with `" ".join` inside of `generate_qc`.
- Then, after the above change, `list2cmdline` would only be used in 1 place (`run_proc`). So, get rid of the function itself, move the functionality directly into `run_proc`.

I didn't mean to suggest using `shlex.quote` in tandem with `" ",join`? Or maybe I'm misunderstanding what you're saying.

Reply:
> If we don't intend/expect the command to be copy-pasted into a terminal, then maybe we can be careless with the quoting.

OH! Of course!! This is a very very good point. I completely missed that the displayed string could be reused in that way.